ui/app/AppLayouts/Wallet/panels/ImportSeedPhrasePanel.qml

```diff
@@ -111,14 +108,12 @@ GridView {
         let words = clipboardText.split(/[, \s]+/)
 
         let timeout = 0
-        if ((grid.model === _internal.twelveWordsModel && words.length === _internal.twentyFourWordsModel) ||
-            (grid.model === _internal.twentyFourWordsModel && words.length === _internal.twelveWordsModel)) {
-            footerItem.pressButton()
-            // Set the teimeout to 100 so the grid has time to generate the new items
-            timeout = 100
-        } else if (words.length !== _internal.twentyFourWordsModel && words.length !== _internal.twelveWordsModel) {
+        let indexOfWordsOption = _internal.seedPhraseWordsOptions.indexOf(words.length)
+        if(indexOfWordsOption == -1) {
             return false
         }
+        footerItem.switchToIndex(indexOfWordsOption)
+        timeout = 100
 
         timer.setTimeout(function(){
```

Comment:
Wondering why the timeout.

Reply:
@stefandunca it was there, so I didn't touch that part, here is the comment which was tehre:
```
// Set the teimeout to 100 so the grid has time to generate the new items
timeout = 100
```
Maybe I should keep it?

Reply:
Sorry, I addressed the wrong question. I meant why go with the delay workaround (to controls not being created) and not directly address the root of the problem (populate the model that generates the controls). Looking a little deeper it seems that would have required some more consideration.